lib/compiler-cranelift/src/config.rs

```diff
@@ -197,6 +188,14 @@ impl CompilerConfig for Cranelift {
         self.enable_verifier = true;
     }
 
+    fn enable_nan_canonicalization(&mut self) {
```

Comment:
Why not an API like `enable_nan_canonicalization(&mut self, enable: bool)`, just like we have for `enable_simd` and probably others?

Reply:
my main concern here is consistency.

On `CompilerConfig` all the other things are enabled through `enable_<something>(&mut self)` functions.

If I decide to add a `enable: bool` parameter to `enable_nan_canonicalization` then I should add the same to all the other `enable_<something>` which is (strictly speaking) a breaking change to a public API that is most likely used a lot.

This is why I introduce a new way to switch this feature on or off. But I agree, the name can be better, something like `toggle_nan_canonicalization(&mut self, enable: bool)` would be better. and then, we can add `enable_<something>(&mut self, enable: bool)`.

WDYT?

Reply:
Let's keep things as they are for the moment.